File: app/src/main/java/com/example/kotlin_notes/MainFragment.java
package com.example.kotlin_notes;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class MainFragment extends Fragment {

    RecyclerView recyclerView;
    NoteAdapter adapter;

    Button add;
    Button sort;
    EditText editText;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_main, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

       add = view.findViewById(R.id.add);
       recyclerView = view.findViewById(R.id.recycler);

        adapter = new NoteAdapter();
        recyclerView.setAdapter(adapter);

        List<Note> list= new ArrayList<>();
        list.add(new Note("","title2","description2",""));
        list.add(new Note("","title3","",""));
        list.add(new Note("","title4","descrip",""));
        list.add(new Note("","title5","description",""));
        adapter.setList(list);


       add.setOnClickListener(v->{
           requireActivity().getSupportFragmentManager()
                   .beginTransaction()
                   .replace(R.id.main_container,new AddFragment())
                   .commit();
       });

       sort= view.findViewById(R.id.sort);
       sort.setOnClickListener(v->{

           String title= editText.getText().toString();
           AddFragment addFragment = new AddFragment();
           Bundle bundle = new Bundle();
           bundle.putString("text",title);
           addFragment.setArguments(bundle);
           requireActivity().getSupportFragmentManager()
                   .beginTransaction()
                   .replace(R.id.main_container,addFragment)
                   .commit();
       });



    }


}

File: app/src/main/java/com/example/kotlin_notes/MainFragment.kt
package com.example.kotlin_notes

import android.app.AlertDialog
import android.content.DialogInterface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView

class MainFragment : Fragment() , NoteAdapter.IOnItem{

    private lateinit var addButon :Button
    private lateinit var  buttonSort : Button
    private lateinit var adapter :NoteAdapter
    private lateinit var searchEdit : EditText
    private lateinit var recyclerView: RecyclerView


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        addButon = view.findViewById(R.id.add)
        buttonSort= view.findViewById(R.id.sort)
        searchEdit = view.findViewById(R.id.search_edit)
        recyclerView= view.findViewById(R.id.recycler)

        adapter= NoteAdapter(this)
        recyclerView.adapter= adapter

        adapter.setList((requireActivity() as MainActivity).list)


        addButon.setOnClickListener{
            val addFragment= AddFragment()
            val fragmentTransaction = parentFragmentManager.beginTransaction()
            fragmentTransaction.replace(R.id.main_container,addFragment).commit()
        }

        buttonSort.setOnClickListener{
            val alertDialog = AlertDialog.Builder(requireContext())
            alertDialog.setTitle("sort")
            alertDialog.setMessage("You want sort by date or titile")
            alertDialog.setPositiveButton("date") {_: DialogInterface,_: Int ->

            }
            alertDialog.setNegativeButton("title"){_:DialogInterface?,_:Int->

            }
        }

    }

    override fun delete(pos: Int) {

    }

    override fun edit(pos: Int, note: NoteModel) {
    }

    override fun share(pos: Int) {
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return LayoutInflater.from(requireContext()).inflate(R.layout.fragment_main,container,false)
    }
}